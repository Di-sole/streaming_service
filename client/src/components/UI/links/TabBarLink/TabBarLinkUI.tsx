import Image from 'next/image';
import Link from 'next/link';
import styles from './TabBarLinkUI.module.scss';

interface TabBarLinkUIProps {
    href: string;
    icon: HTMLImageElement;
    text: string;
    selected: boolean;
}

const TabBarLinkUI = ({ href, icon, text, selected }: TabBarLinkUIProps) => {
    return (
        <Link
            id="tab-item"
            href={href}
            className={[styles.container, selected ? styles.container_selected : ''].join(' ')}
        >
            <div className={styles.container__itemGlow}></div>
            <div className={styles.container__itemIcon}>
                <Image src={icon} height={20} width={20} alt={`Иконка ${text}`} />
            </div>
            <div className={styles.container__itemCaption}>{text}</div>
        </Link>
    );
};

export default TabBarLinkUI;
