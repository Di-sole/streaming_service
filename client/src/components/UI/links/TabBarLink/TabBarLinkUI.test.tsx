import React from 'react';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import TabBarLinkUI from './TabBarLinkUI';
import catalogueIcon from '../../../../../public/icons/tabbar_icons/catalogue.svg';

describe('TabBarLinkUI SNAPSHOT TEST', () => {
    test('TabBarLinkUI should not be changed', () => {
        render(<TabBarLinkUI selected={true} href="" icon={catalogueIcon} text="Каталог" />);
        const link = screen.getByRole('link');

        expect(link).toMatchSnapshot();
    });
});
