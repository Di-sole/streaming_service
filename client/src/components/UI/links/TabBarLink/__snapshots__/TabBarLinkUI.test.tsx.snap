// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabBarLinkUI SNAPSHOT TEST TabBarLinkUI should not be changed 1`] = `
<a
  class="container container_selected"
  href=""
  id="tab-item"
>
  <div
    class="container__itemGlow"
  />
  <div
    class="container__itemIcon"
  >
    <img
      alt="Иконка Каталог"
      data-nimg="1"
      decoding="async"
      height="20"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style="color: transparent;"
      width="20"
    />
  </div>
  <div
    class="container__itemCaption"
  >
    Каталог
  </div>
</a>
`;
